eventlet/wsgi.py

```diff
@@ -456,6 +448,8 @@ def start_response(status, response_headers, exc_info=None):
                 minimum_write_chunk_size = int(self.environ.get(
                     'eventlet.minimum_write_chunk_size', self.minimum_chunk_size))
                 for data in result:
+                    if not isinstance(data, six.binary_type):
```

Comment:
This is a huge API change and may break existing applications. The only way I see it included in v0.17 is keep previous behavior by default and enable bytes check with environment flag, like `EVENTLET_WSGI_BYTES`.